src/context.rs

```diff
@@ -192,15 +195,41 @@ pub struct Context<'a, 'b: 'a, 's> {
 
 ///A reader for a request body.
 pub struct BodyReader<'a, 'b: 'a> {
-    request: HttpReader<&'a mut BufReader<&'b mut NetworkStream>>
+    reader: HttpReader<&'a mut BufReader<&'b mut NetworkStream>>,
+    is_multipart: bool,
```

Comment:
The boundary is pretty integral to the multipart specification. Even just a single field has to have a boundary before and after it. I guess my API could be simplified to just the one method.